backend/views/syson-diagram-statetransition-view/src/main/java/org/eclipse/syson/diagram/statetransition/view/nodes/DefinitionNodeDescriptionProvider.java

```diff
@@ -41,9 +41,9 @@ public DefinitionNodeDescriptionProvider(EClass eClass, IColorProvider colorProv
     protected List<NodeDescription> getReusedChildren(IViewDiagramElementFinder cache) {
         var reusedChildren = new ArrayList<NodeDescription>();
 
-        StateTransitionViewDiagramDescriptionProvider.COMPARTMENTS_WITH_LIST_ITEMS.forEach((type, listItems) -> {
+        StateTransitionViewDiagramDescriptionProvider.COMPARTMENTS_WITH_MERGED_LIST_ITEMS.forEach((type, listItems) -> {
             if (type.equals(this.eClass)) {
-                listItems.forEach(eReference -> {
+                listItems.stream().findFirst().ifPresent(eReference -> {
```

Comment:
It seems strange you have to change the `forEach` for `stream().findFirst()`.
Why did you do that?

Reply:
I thought at the beggining it was necessary but finally it is not. Thanks.